src/Symfony/Component/String/AbstractUnicodeString.php

```diff
@@ -396,6 +396,14 @@ public function trimStart(string $chars = " \t\n\r\0\x0B\x0C\u{A0}\u{FEFF}"): pa
         return $str;
     }
 
+    public function underscore(): parent
+    {
+        $str = $this->camel()->snake();
+        $str->string = mb_strtolower(preg_replace('~(?<=[a-z])([0-9])~u', '_$1', $str->string), 'UTF-8');
```

Comment:
Why doing ` camel -> snake -> regexp replacement -> lowercase`, where `snake` itself is already doing `camel -> title -> regex replacement ->lowercase` with a regex keeping numbers in the previous word ? This duplicates the camelcase conversion, performs 2 regexp replacement where one could be enough, and converts to lowercase twice.

Reply:
Currently doing `camel -> title -> regex -> lowercase`. Good?